### PR TITLE
Karknu/rcd 43 mitigation

### PR DESCRIPTION
**Pull request by karknu**
## Description

Improvement that makes the network disconnect in RCD-43 less likely to happen.

## Type of change

- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [ ] 🛠 New feature (non-breaking change which adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🏭 Refactoring that does not change existing functionality but does improve things like code readability, structure etc
- [ ] 🔨 New or improved tests for existing code
- [ ] ⛑ git-flow chore (backport, hotfix, etc)

## Developer checklist

- [ ] I have read the [style guide](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/style-guide.md) document, and my code follows the code style of this project.
- [ ] If my code deals with exceptions, it follows the [guidelines](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/exceptions.md).
- [ ] I have updated any documentation accordingly, if needed. Documentation changes can be reflected in opening a PR on [cardanodocs.com](https://github.com/input-output-hk/cardanodocs.com), amending the inline [Haddock](https://www.haskell.org/haddock/) comments, any relevant README file or one of the document listed in the [docs](https://github.com/input-output-hk/cardano-sl/tree/develop/docs) directory.
- [ ] CHANGELOG entry has been added and is linked to the correct PR on GitHub.

## Testing checklist

- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

## How to merge

Send the message `bors r+` to merge this PR. For more information, see
[`docs/how-to/bors.md`](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/how-to/bors.md).

### CODE REVIEW

File: db/src/Pos/DB/Rocks/Functions.hs [hunk 1]
```diff
@@ -91,6 +93,12 @@ openNodeDBs recreate fp = do
         , lrcPath
         , miscPath
         ]
+
+    when (System.Info.os == "darwin") $ do
+        -- Prevent indexing of blocks on OSX
+        _ <- openFile (fp </> ".metadata_never_index") AppendMode
+        pure ()
```
Comment (dcoutts) [inline db/src/Pos/DB/Rocks/Functions.hs]: Why not just `writeFile` ? Or if you want to do `openFile`, at least we should `hClose` too.

Comment (dcoutts) [reply db/src/Pos/DB/Rocks/Functions.hs]: Or
```
withFile (fp </> ".metadata_never_index") AppendMode (\_ -> return ())
```

Comment (cleverca22) [reply db/src/Pos/DB/Rocks/Functions.hs]: also, what about
```
import Distribution.System

when (buildOS == OSX) $ do
```
sum types are less likely to have mistakes then string compares

Comment (dcoutts) [reply db/src/Pos/DB/Rocks/Functions.hs]: That's true, and that's why Cabal implements that interface, but we can't do the same without depending on Cabal.

Comment (cleverca22) [reply db/src/Pos/DB/Rocks/Functions.hs]: we already depend on Cabal for the SystemTag logic in the update system and in a dozen places for the current version from the cabal file

**Comment by jmitchell** [comment]
```
src/Pos/DB/Rocks/Functions.hs:97:5: error:
    • Couldn't match type ‘m’ with ‘IO’
      ‘m’ is a rigid type variable bound by
        the type signature for:
          openNodeDBs :: forall (m :: * -> *).
                         MonadIO m =>
                         Bool -> FilePath -> m NodeDBs
        at src/Pos/DB/Rocks/Functions.hs:(72,1)-(74,36)
      Expected type: m ()
        Actual type: IO ()
    • In a stmt of a 'do' block:
        when (System.Info.os == "darwin")
          $ withFile
              (fp </> ".metadata_never_index") AppendMode (\ _ -> return ())
      In the expression:
        do liftIO
             $ whenM ((recreate &&) <$> doesDirectoryExist fp)
                 $ removeDirectoryRecursive fp
           let blocksDir = fp </> "blocks"
           let blocksIndexPath = blocksDir </> "index"
           let _blockDataDir = blocksDir </> "data"
           ....
      In an equation for ‘openNodeDBs’:
          openNodeDBs recreate fp
            = do liftIO
                   $ whenM ((recreate &&) <$> doesDirectoryExist fp)
                       $ removeDirectoryRecursive fp
                 let blocksDir = ...
                 let blocksIndexPath = ...
                 ....
            where
                ensureDirectoryExists :: MonadIO m => FilePath -> m ()
                ensureDirectoryExists = liftIO . createDirectoryIfMissing True
    • Relevant bindings include
        openNodeDBs :: Bool -> FilePath -> m NodeDBs
          (bound at src/Pos/DB/Rocks/Functions.hs:75:1)
   |
97 |     when (System.Info.os == "darwin") $
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^...
```

**Pull request merged**